python/dgl/graphbolt/base.py

```diff
@@ -5,6 +5,15 @@
 
 import torch
 from torch.torch_version import TorchVersion
+
+if TorchVersion(torch.__version__) >= "2.3.0":
+    # Due to https://github.com/dmlc/dgl/issues/7380, we need to check if dill
```

Comment:
Add TODO, and file an issue to track.

Reply:
This is already resolved in nightly torch builds.

Reply:
https://github.com/pytorch/pytorch/issues/124907

Reply:
Maybe we can make it `==2.3.0` instead, because it looks like the fix will most likely be available in 2.3.1.

Reply:
we don't know when or whether 2.3.1 will be released, maybe they just release 2.4.0 though it's unlikely. So I didn't involve `2.3.1` here explicitly

Reply:
I see. I think they are releasing 2.3.1 for sure. https://github.com/pytorch/pytorch/issues/125425

Reply:
But I am not fully sure if the fix will be included as part of 2.3.1 or not.